ext4: Initialize writeback_index to 0 when allocating a new inode

From: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>

The write_cache_pages() function uses the mapping->writeback_index as
the starting index to write out when range_cyclic is set.  Properly
initialize writeback_index so that we start the writeout at index 0.

This was found when debugging the small file fragmentation on ext4.

Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---

 fs/ext4/super.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

Index: linux-2.6.27-rc3/fs/ext4/super.c
===================================================================
--- linux-2.6.27-rc3.orig/fs/ext4/super.c	2008-08-18 11:49:32.000000000 -0700
+++ linux-2.6.27-rc3/fs/ext4/super.c	2008-08-18 11:49:39.000000000 -0700
@@ -568,6 +568,7 @@ static struct inode *ext4_alloc_inode(st
 #endif
 	ei->i_block_alloc_info = NULL;
 	ei->vfs_inode.i_version = 1;
+	ei->vfs_inode.i_data.writeback_index = 0;
 	memset(&ei->i_cached_extent, 0, sizeof(struct ext4_ext_cache));
 	INIT_LIST_HEAD(&ei->i_prealloc_list);
 	spin_lock_init(&ei->i_prealloc_lock);
